test/plugin_functional/test_suites/core_plugins/history_block.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { PluginFunctionalProviderContext } from '../../services';
+
+export default function ({ getService, getPageObjects }: PluginFunctionalProviderContext) {
+  const PageObjects = getPageObjects(['common']);
+  const browser = getService('browser');
+  const testSubjects = getService('testSubjects');
+
+  describe('application using `ScopedHistory.block`', () => {
+    beforeEach(async () => {
+      await PageObjects.common.navigateToApp('core_history_block');
+    });
+
+    describe('when navigating to another app', () => {
+      it('prevents navigation if user click cancel on the confirmation dialog', async () => {
+        await testSubjects.click('applink-external-test');
+
+        await testSubjects.existOrFail('navigationBlockConfirmModAl');
+        await PageObjects.common.clickCancelOnModal(false);
+        expect(await browser.getCurrentUrl()).to.contain('/app/core_history_block');
+      });
+      it('allows navigation if user click confirm on the confirmation dialog', async () => {
+        await testSubjects.click('applink-external-test');
+
+        await testSubjects.existOrFail('navigationBlockConfirmModAl');
+        await PageObjects.common.clickConfirmOnModal();
+        expect(await browser.getCurrentUrl()).to.contain('/app/home');
+      });
+    });
+
+    describe('when navigating to another app', () => {
+      it('prevents navigation if user click cancel on the confirmation dialog', async () => {
+        await testSubjects.click('applink-intra-test');
+
+        await testSubjects.existOrFail('navigationBlockConfirmModAl');
+        await PageObjects.common.clickCancelOnModal(false);
+        expect(await browser.getCurrentUrl()).to.contain('/app/core_history_block');
+        expect(await browser.getCurrentUrl()).not.to.contain('/foo');
+      });
+      it('allows navigation if user click confirm on the confirmation dialog', async () => {
+        await testSubjects.click('applink-intra-test');
+
+        await testSubjects.existOrFail('navigationBlockConfirmModAl');
+        await PageObjects.common.clickConfirmOnModal();
+        expect(await browser.getCurrentUrl()).to.contain('/app/core_history_block/foo');
+      });
```

Comment:
NIT: What happens if we navigate back to `core_history_block`, do we get the confirmation again? Can we add a test to ensure we don't leave hanging Prompts? How about re-entering the app? Will it re-register the Prompt a 2nd time?

Reply:
> What happens if we navigate back to core_history_block, do we get the confirmation again? 

In this specific test plugin's implementation, we don't, as we are using the `Prompt` component on the `/` page only, so the block handler will be disposed when leaving this specific page. I can add a test asserting this behavior, even if it's more the plugin's implementation detail we would be testing here.

> Can we add a test to ensure we don't leave hanging Prompts? How about re-entering the app? Will it re-register the Prompt a 2nd time?

When re-entering the app, a new `ScopedHistory` instance will be created, and the Prompt will be reattach from this new instance to the global history. The block handlers removal when the history changes from active to inactive is more heavily tested in the ScopedHistory's unit tests (which is way easier than from FTR).